src/snapshot-editor/src/info.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+
+use clap::Subcommand;
+use vmm::persist::MicrovmState;
+
+use crate::utils::*;
+
+#[derive(Debug, thiserror::Error)]
+pub enum InfoVmStateError {
+    #[error("{0}")]
+    Utils(#[from] UtilsError),
+}
+
+#[derive(Debug, Subcommand)]
+pub enum InfoVmStateSubCommand {
+    /// Print info about vcpu states.
+    #[cfg(target_arch = "aarch64")]
+    VcpuStates {
+        /// Path to the vmstate file.
+        #[arg(short, long)]
+        vmstate_path: PathBuf,
+    },
+}
+
+pub fn info_vmstate_command(command: InfoVmStateSubCommand) -> Result<(), InfoVmStateError> {
+    match command {
+        #[cfg(target_arch = "aarch64")]
+        InfoVmStateSubCommand::VcpuStates { vmstate_path } => {
+            info(&vmstate_path, info_vcpu_states)?
+        }
+    }
+    Ok(())
+}
+
+fn info(
+    vmstate_path: &PathBuf,
+    f: impl Fn(&MicrovmState) -> Result<(), InfoVmStateError>,
```

Comment:
Same as before, I think it's much simpler to just have a single function here that opens the file and prints the related info.

Reply:
Ok, the previous case is fine. But here, will we ever do something with print info command other than printing info?

Reply:
hmm... printing more info? I thought of adding print command for each data type in vmstate: vm memory, device states... etc.